morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -88,13 +166,74 @@ instance
       MorphoStdHeader {..} = morphoHeaderStd
 
 instance MPretty MorphoBlockTx where
-  mpretty = mpretty . morphoBlockGenTx
+  mpretty (MorphoBlockTx genTx genTxId) =
+    "MorphoBlockTx(transaction:"
+      <+> mpretty genTx
+      <> ", transactionId:"
+      <+> mpretty genTxId
+      <> ")"
+
+instance MPretty (ChainHash (MorphoBlock h c)) where
+  mpretty GenesisHash = "GenesisHash"
+  mpretty (BlockHash h) = "BlockHash(" <> mpretty h <> ")"
+
+instance MPretty (HeaderHash a) => MPretty (Tip a) where
+  mpretty TipGenesis = "TipGenesis"
+  mpretty (Tip s h b) =
+    "Tip(slot:"
+      <+> mpretty s
+      <> ", hash:"
+      <+> mpretty h
+      <> ", blockNo:"
+      <+> mpretty b
+      <> ")"
 
 instance MPretty Tx where
   mpretty (Tx vote) = mpretty vote
 
 instance MPretty Vote where
-  mpretty = viaShow
+  mpretty (Vote block signature) =
+    let bytes = powBlockRefToBytes block
+        pubKey = recoverPublicKey signature bytes
+     in "Vote(block:"
+          <+> mpretty block
+          <> ", byPubkey:"
+          <+> mpretty pubKey
```

Comment:
Currently this is

> Vote(block: PowBlock(number: 4, hash: 1b3eb035), byPubkey: Just(PublicKey(ec33a368))

Let's get rid of the `Just` the same way as for `MPretty Checkpoint`, showing `Nothing` as `\<unknown\>`.